### DISCUSSION TITLE
Determining TRE Order

### DISCUSSION THREAD

**Discussion by eyanko** [mdaus/nitro#207] [Q&A]
I am using the C++ version and need to be able to remove a TRE, recreate the TRE, and then place the newly created TRE back in the same place as the original TRE. Currently, I am removing the TRE by name, creating the new TRE, and then appending the TRE back to the extended header section, but this places the TRE at the end of the list. For example, if the TRE is the third in the list out of five, the current method puts it back in fifth place, but ideally I would like to put it back in third. I didn't see any C++ methods for determining the position of a TRE nor placing it in a specific position.  Is this possible? I did see the Iterator, so perhaps that would be of use if the Iterator always returns the TREs in the same order?

Thanks!

**Comment by bradh** [comment]
I don't know the answer, but I wonder why the order matters. In general, TREs should be able to be interpreted in any order.

**Comment by eyanko** [comment]
In general I would agree, but our product specs specify an order for this case. The original element we are modifying has the correct order, and we need the new element with the updated TRE to match.
